### PR TITLE
Jmprieur/msal3.0.3 jmpreview

### PR DESCRIPTION
**Pull request by jmprieur**
## Purpose
SImplifing the code by factorizing out reusable code

## Does this introduce a breaking change?
```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature (to the library)
[ ] Code style update (formatting, local variables)
[ x] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
```

* Test the code
Follow the readme.md
```
```

### CODE REVIEW

File: 1. Desktop app calls Web API/TodoListService/Startup.cs [hunk 1]
```diff
@@ -46,23 +46,9 @@ public Startup(IConfiguration configuration)
...
-                options.Authority += "/v2.0";
-				
-				// The valid audiences are both the Client ID (options.Audience) and api://{ClientID}
-                options.TokenValidationParameters.ValidAudiences = new string[] { options.Audience, $"api://{options.Audience}" };
-
-                // Instead of using the default validation (validating against a single tenant, as we do in line of business apps),
-                // we inject our own multitenant validation logic (which even accepts both V1 and V2 tokens)
-                options.TokenValidationParameters.IssuerValidator = AadIssuerValidator.ForAadInstance(options.Authority).ValidateAadIssuer;
+            services.AddProtectWebApiWithMicrosoftIdentityPlatformV2(Configuration)
+                    ;
```
Comment (bgavrilMS) [inline 1. Desktop app calls Web API/TodoListService/Startup.cs]: ; on newline

Comment (jmprieur) [reply 1. Desktop app calls Web API/TodoListService/Startup.cs]: Thanks @bgavrilMS. Will address in next pr

File: 2. Web API now calls Microsoft Graph/TodoListService/TodoListService.csproj [hunk 2]
```diff
@@ -10,7 +10,11 @@
     <PackageReference Include="Microsoft.AspNetCore.App" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.AzureAD.UI" Version="2.2.0" />
     <PackageReference Include="Microsoft.AspNetCore.Razor.Design" Version="2.2.0" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.Identity.Client" Version="2.7.1" />
+    <PackageReference Include="Microsoft.Identity.Client" Version="3.0.3-jmpreview" />
```
Comment (brentschmaltz) [inline 2. Web API now calls Microsoft Graph/TodoListService/TodoListService.csproj]: A common practice is to have a 'dependencies.prop' file that defines the dependencies version.
For example the file would contain:
<MicrosoftIdentityClientVersion>3.0.3</MicrosoftIdentityClientVersion>

Then TodoListService.csproj would include dependencies.prop and write:
<PackageReference Include="Microsoft.Identity.Client" Version="$(MicrosoftIdentityClientVersion)"/>

This makes updating versions simpler and the same across projects.
For example see: 
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev5x/build/dependencies.props

https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev5x/src/Microsoft.IdentityModel.Tokens/Microsoft.IdentityModel.Tokens.csproj

Comment (jmprieur) [reply 2. Web API now calls Microsoft Graph/TodoListService/TodoListService.csproj]: Thanks Brent. We'll do that.

**Pull request merged**